filter_plugins/cisco_nxos.py

```diff
@@ -0,0 +1,444 @@
+            break
+
+    return vlan_list
+
+def removeOldLines(config: str, interfaces: dict, vlans: dict):
+    """
+    This method will take a NXOS config block and remove all the lines that should be overwritten by the ansible generated config
+    This includes any interface block defined in the config that is in the manifest, except those with "managed": true in the manifest
+
+    Example:
+
+    interface Ethernet1/33
+        shutdown
+
+    interface Ethernet1/34
+        shutdown
+
+    interface Ethernet1/35
+        shutdown
+
+    interface Ethernet1/36
+        shutdown
+
+    interface mgmt0
+      vrf member management
+      ip address 10.80.3.1/20
+    icam monitor scale
+
+    line console
+    line vty
+    boot nxos bootflash:/nxos64-cs.10.3.2.F.bin
+    no system default switchport shutdown
+
+    Output of this method will be:
+
+    interface mgmt0
+      vrf member management
+      ip address 10.80.3.1/20
+    icam monitor scale
+
+    line console
+    line vty
+    boot nxos bootflash:/nxos64-cs.10.3.2.F.bin
+    no system default switchport shutdown
+
+    :param config: The NXOS config block to parse
+    :type config: str
+    :param interfaces: The interfaces manifest
+    :type interfaces: dict
+    :param vlans: The VLANs manifest
+    :type vlans: dict
+    :return: The existing running config without lines that are maintained by ansible manifest
+    :rtype: str
+    """
+    config = config.splitlines()
+    new_config = []
+
+    # Remove lines that will be filled in by this plugin
+    # When delete section is true lines will be deleted
+    # The idea is that after a section is over (such as an interface block)
+    # delete section will be reset to false and be put back to true only if
+    # the next line is determined to be a section that should be deleted
+    delete_section = False
+    found_vlanrange_line = False
+    for line in config:
+        if line.startswith("!"):
+            # skip commands
+            continue
+
+        if line == "" and delete_section:
+            # Skip empty lines if deleting
+            continue
+
+        if not line.startswith(" "):
+            # This is a new section
+            delete_section = False
+
+        if line.startswith("interface breakout "):
+            # Existing breakout config, get port
+            if match := re.search(r"port (\d+)", line):
+                port_num = match.group(1)
+                object_name = f"Ethernet1/{port_num}"
+                if object_name in interfaces:
+                    continue
+
+        if line.startswith("interface "):
+            # This is an interface config
+
+            # Find the object name
+            object_name = line.removeprefix("interface ")
+            if not object_name.startswith("Ethernet") and not object_name.startswith("port-channel"):
+                # This allows things like management interfaces to pass through
```

Comment:
wouldn't this also allow vlan interfaces to passthrough? I assumed that we want to manage the vlan interfaces with these manifests?

Reply:
as far as I can tell nxos doesn't have vlan interfaces, everything is configured on the vlan itself, so it would show up as `vlan XYZ` in the config with `interface` in the beginning

Reply:
It does support vlan interfaces. Our older cisco switches support it so that's why I was expecting it. But the same commands did not run on the new cisco switches, so I looked up [the documentation](https://www.cisco.com/c/en/us/td/docs/switches/datacenter/nexus9000/sw/7-x/interfaces/configuration/guide/b_Cisco_Nexus_9000_Series_NX-OS_Interfaces_Configuration_Guide_7x/b_Cisco_Nexus_9000_Series_NX-OS_Interfaces_Configuration_Guide_7x_chapter_0101.html#task_F0C90E46B5D84349BB62D68BC466B7CB) and then I learnt that we have to enable vlan interfaces first, and then it works as expected.

I enabled it with `feature interface-vlan` on the NERC-CORE-1 and then you can create vlan interfaces. Here's an example I created:

```
NERC-CORE-1(config)# feature inter
interface-vlan   intersight
NERC-CORE-1(config)# feature interface-vlan ?
  <CR>

NERC-CORE-1(config)# feature interface-vlan
NERC-CORE-1(config)# inter
interface    intersight
NERC-CORE-1(config)# interface vlan 201?
  ,         Multi range separator
  -         Range separator
  <1-4094>  Vlan interface number

NERC-CORE-1(config)# interface vlan 201
```

as soon as I enabled it, it also created a default interface for vlan 1.

```
NERC-CORE-1# show running-config |grep "interface Vlan"
interface Vlan1
interface Vlan201
```

feel free to delete the vlan interfaces or disable it for now.